liquidprompt

```diff
@@ -3390,10 +3448,20 @@ _lp_default_theme_prompt_template() {
         # add title escape time, jobs, load and battery
         PS1="${LP_PS1_PREFIX}${LP_TIME}${LP_BATT}${LP_LOAD}${LP_TEMP}${LP_WIFI}${LP_JOBS}"
         # add user, host, permissions colon, working directory, and dirstack
-        PS1+="${LP_BRACKET_OPEN}${LP_USER}${LP_HOST}${LP_PERM}${LP_PWD}${LP_DIRSTACK}${LP_BRACKET_CLOSE}${LP_SHLVL}"
+        PS1+="${LP_BRACKET_OPEN}${LP_USER}${LP_HOST}${LP_PERM}${LP_PWD}${LP_DIRSTACK}${LP_BRACKET_CLOSE}${LP_PROXY}"
 
         # add development environments
-        PS1+="${LP_SCLS}${LP_AWS_PROFILE}${LP_CONTAINER}${LP_VENV}${LP_NODE_VENV}${LP_RUBY_VENV}${LP_TFSPACE}${LP_KUBECONTEXT}${LP_PROXY}"
+        local dev_all=("${LP_SCLS}" "${LP_AWS_PROFILE}" "${LP_CONTAINER}" "${LP_VENV}" "${LP_NODE_VENV}" "${LP_RUBY_VENV}" "${LP_TFSPACE}" "${LP_KUBECONTEXT}" "${LP_CMAKE}")
+        local dev_active=()
+        for i in "${dev_all[@]}"; do
+            if [[ "$i" ]]; then
+                dev_active+=("$i")
+            fi
+        done
+        if [[ ${#dev_active[*]} -gt 0 ]] ; then
+            _lp_join "${LP_MARK_DEV_MID}" "${dev_active[@]}"
+            PS1+=" ${LP_MARK_DEV_OPEN}${lp_join}${LP_MARK_DEV_CLOSE}"
+        fi
```

Comment:
This should be broken out into a function, maybe `_lp_dev_env_color()`.

It also should be assigned to a variable like `$LP_DEV_ENVS`, so those who want to use it this new way but in a different spot can do so. And good on keeping the original `$LP_SCLS` and such.